Document/0x05e-Testing-Cryptography.md

```diff
@@ -232,15 +253,18 @@ public static SecretKey generateStrongAESKey(char[] password, int keyLength)
     return new SecretKeySpec(keyBytes, "AES");
 }
 ```
-The above method requires a character array containing the password and the needed keylength in bits, for instance a 128 or 256-bit AES key. We define an iteration count of 10000 rounds which will be used by the PBKDFv2 algorithm. This significantly increases the workload for a bruteforce attack. We define the salt size equal to the key length, we divide by 8 to take care of the bit to byte conversion. We use the ```SecureRandom``` class to randomly generate a salt. Obviously, the salt is something you want to keep constant to ensure the same encryption key is generated time after time for the same supplied password. Storing the salt does not need any additional security measures, this can be publicly stored in the ```SharedPreferences``` without the need of any encryption. Afterwards the Password-based Encryption (PBE) key is generated using the recommended ```PBKDF2WithHmacSHA1``` algorithm.
+The above method requires a character array containing the password and the needed keylength in bits, for instance a 128 or 256-bit AES key. We define an iteration count of 10000 rounds which will be used by the PBKDFv2 algorithm. This significantly increases the workload for a bruteforce attack. We define the salt size equal to the key length, we divide by 8 to take care of the bit to byte conversion. We use the ```SecureRandom``` class to randomly generate a salt. Obviously, the salt is something you want to keep constant to ensure the same encryption key is generated time after time for the same supplied password. Storing the salt does not need any additional security measures, this can be privately stored in the ```SharedPreferences``` without the need of any encryption. Note that this will allow for device synchronization if you do not exclude it from the Android backup. See "testing android storage" for more details. Afterwards the Password-based Encryption (PBE) key is generated using the recommended ```PBKDF2WithHmacSHA1``` algorithm till API version 26. From there on it is best to use ```PBKDF2withHmacSHA256```, which will end up with a different keysize.
```

Comment:
keylength -> key length
10000 -> 10 000 | 10,000 | 10.000 ? 
PBKDFv2 -> PBKDF2 (as spelled in rfc2898)
till API version 26 -> 
Alt.1 : up until API version 26
Alt.2 : For API 26 and below
From there on it is best to use -> For API versions above 26, it is best to use
which will end up with a different keysize -> which will result in a different keysize